scripts/update_deps.py

```diff
@@ -482,6 +482,15 @@ def CMakeConfig(self, repos):
                     var_name=d['var_name'],
                     install_dir=dep_commit[0].install_dir))
 
+        # Override google test options from the known-good.json with ones from the command line
+        if self.name == "googletest" and (self._args.gtest_shared_libs or self._args.gtest_force_shared_crt):
+            for index,option in enumerate(self.cmake_options):
+                if self._args.gtest_shared_libs and 'build_shared_libs' in option.lower():
+                    self.cmake_options[index] = f"-DBUILD_SHARED_LIBS={str(self._args.gtest_shared_libs)}"
+                    print(f'INFO: Setting googletest option: -DBUILD_SHARED_LIBS={str(self._args.gtest_shared_libs)}')
+                if self._args.gtest_force_shared_crt and 'gtest_force_shared_crt' in option.lower():
+                    self.cmake_options[index] = f"-Dgtest_force_shared_crt={str(self._args.gtest_force_shared_crt)}"
+                    print(f'INFO: Setting googletest option: -Dgtest_force_shared_crt={str(self._args.gtest_force_shared_crt)}')
```

Comment:
Those options are actually broken in googletest currently, doing nothing to change the CRT used.

https://github.com/google/googletest/issues/4731 is an open issue documenting it that and I added my findings about it.

The actual review comments I have is that the code looks reasonable, and there isn't any reason to reject the change just because googletest doesn't currently make use of it.

Reply:
I think the options we want is:

```
BUILD_SHARED_LIBS=OFF
gtest_force_shared_crt=ON
```

We can delete `"-Dgtest_force_shared_crt=ON",` from known_good.json to get that, as OFF is the default value for googletest 1.14. BUILD_SHARED_LIBS is already set to off and can be left alone.

Reply:
It's been difficult removing `gtest_force_shared_crt` from known-good.json, which otherwise forces it to "off." That seems to break running ctest on a non-static build of Qt, which also breaks github actions. I've decided to leave those options for update_deps.sh in and let user's control it.